mycroft/skills/common_play_skill.py

```diff
@@ -88,7 +107,13 @@ def __handle_play_query(self, message):
                                         "searching": True}))
 
         # Now invoke the CPS handler to let the skill perform its search
-        result = self.CPS_match_query_phrase(search_phrase)
+        # TODO remove try: except: block in major release XXX
+        # it is needed for old skills which are not expecting media_type
+        # alternatively check for function signature before calling it
+        try:
+            result = self.CPS_match_query_phrase(search_phrase, media_type)
+        except:
```

Comment:
to be specific
```
except TypeError
```

This will have the effect that the method it's called twice if a TypeError is raised but better than twice if any Exception occurs.

To go the inspect path something like

```python
from inspect import signature

[...]

    if len(signature(self.CPS_match_query_phrase)) == 2:
        result = self.CPS_match_query_phrase(search_phrase, media_type)
    else:
        result = self.CPS_match_query_phrase(search_phrase)
```